build.rs

```diff
@@ -131,3 +138,14 @@ fn rustc_version() -> Option<RustcVersion> {
         nightly: nightly,
     })
 }
+
+fn proc_macro_span_disallowed() -> bool {
+    if let Some(rustflags) = env::var_os("RUSTFLAGS") {
+        for flag in rustflags.to_string_lossy().split(" ") {
```

Comment:
As far as I can tell Cargo/rustc don't allow any escaped whitespace in this flag, it must either be `-Z allow-features=foo,bar` or `-Zallow-features=foo,bar`, so simply splitting on whitespace and checking for string-inclusion should be correct.

There is the chance of a user finding some other flag that they can pass both strings `allow-features` and `proc_macro_span` in, but that seems rare enough that trying to 100% correctly parse these flags isn't worthwhile.